### PR TITLE
Add field collections via campaign JSON.

### PR DESCRIPTION
**Pull request by DFurnes**
## Changes
- Ability to add arrays of field collections (FAQs, tips sections).

For review: @aaronschachter @angaither 
## Screenshot

campaign-with-field-collections

### CODE REVIEW

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.helpers.inc [hunk 1]
```diff
@@ -391,6 +391,18 @@ function dosomething_campaign_create_node_from_json($string) {
     }
   }
 
+  // Set all field collection properties:
+  $field_collection = dosomething_campaign_get_property_names_field_collection();
+  foreach($field_collection as $field_collection_field) {
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.helpers.inc]: Not needed for merge but pretty sure `foreach` should have a space after it.  https://www.drupal.org/coding-standards#controlstruct

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.helpers.inc [hunk 2]
```diff
@@ -463,6 +475,37 @@ function dosomething_campaign_get_property_names_image() {
...
+function dosomething_campaign_create_field_collection_from_json($string, $field_collection_name, $node) {
+  $data = json_decode($string);
+
+  $field_collection_item = entity_create('field_collection_item', array('field_name' => $field_collection_name));
+  $field_collection_item->setHostEntity('node', $node);
+
+  foreach($data as $data_field => $data_field_value) {
+    $collection_field_name =  "field_compound_text_" . $data_field;
+    $field_collection_item->{$collection_field_name}[LANGUAGE_NONE][0]['value'] = $data_field_value;
+    if($data_field == "copy") {
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.helpers.inc]: Same, spacing. https://www.drupal.org/coding-standards#controlstruct

**Comment by aaronschachter** [comment]
Just some picky picky spacing. Otherwise :+1:

File: tests/fixtures/campaign.json [hunk 3]
```diff
@@ -74,9 +88,25 @@
 
     "pre_step_header": "Get your 'Fee!",
     "pre_step_copy": "You'll need to find some 'fee, fast. Find Maxwell Watson.",
+    "step_pre": [
+      {
+        "header": "Find Maxwell",
```
Comment (angaither) [inline tests/fixtures/campaign.json:93]: this is not future proof. :trollface:

**Comment by angaither** [comment]
:shipit:

**Pull request merged**